README.md

```diff
@@ -0,0 +1,175 @@
+        "__delta" : {
+            "__v" : 0,
+            "name" : "Rover",
+            "_id" : ObjectId("5aadcc855df61e1a63490c67")
+        },
+        "__logBehaviour" : "delta"
+    },
+    "objectType" : "Dog",
+    "action" : "updated",
+    "actor" : "bob@exmple.com",
+    "when" : ISODate("2018-03-18T02:18:46.101Z"),
+    "attributes" : null,
+    "callStack" : [ 
+        "src/index.js:10:20"
+    ]
+}
+```
+
+
+## Configuration
+
+### Logger Configuration
+When initialising the logger, the database is configurable
+```js
+Logger.init({
+  url: `mongodb://localhost:27017/event-log`,
+  collection: 'events'
+})
+```
+
+| Variable      | Description   
+| ------------- |:-------------|
+| url           | The url of the mongo database to log the events to. It is recommended to use a different database to your application's db|
+| collection    | The name of the collection to store events in |
+
+
+
+### Model Configuration
+The event logger allows different logging behaviour to be configured based on the type of database modification (create, update, delete) and /or based on the actor that performed the action (system process or user).
+
+
+```js
+await new Dog({ name: 'Cookie' }).by('system').save(); // logs with system behaviour
+await new Dog({ name: 'Rover' }).by('john@example.com').save(); // logs with user behaviour
+```
+
+
+#### Logging Behaviour
+* **delta**: Only stores the difference between the two objects, i.e. the properties that were actually changed. 
+* **snapshot**: Stores a snapshot of the entire object at the time it was updated
+* **snapshotAndDelta**: Stores the snapshot as well as the difference form pre-update to post-update
+* **id**: Only stores the id of the object which was changed
+
+
+##### Default Behaviour
+```
+// If changed by a system actor
+`if.${Action.created}.by.${Actor.system}` = Behaviour.snapshot;
+`if.${Action.updated}.by.${Actor.system}` = Behaviour.delta;
+`if.${Action.deleted}.by.${Actor.system}` = Behaviour.snapshot;
+
+// If changed by a user actor
+`if.${Action.deleted}.by.${Actor.user}` = Behaviour.snapshot;
+`if.${Action.created}.by.${Actor.user}` = Behaviour.snapshot;
+`if.${Action.updated}.by.${Actor.user}` = Behaviour.delta;
+```
+
+##### Setting the Config
+
+```js
+  import { Actor, Behaviour, Action } from 'mongoose-events-logger';
+
+  const options = Dog.loggingOptions();
+  options.if[Action.updated].by[Actor.user] = Behaviour.snapshotAndDelta;
+  options.if[Action.created].by[Actor.user] = Behaviour.snapshotAndDelta;
+  options.if[Action.deleted].by[Actor.user] = Behaviour.snapshotAndDelta;
+  Dog.setLoggingOptions(options);
+```
+
+
+#### Ignore attributes
+You can optionally choose to ignore specific attributes on your model that will be ignored by the events logger when performing updates on a model. 
+
+
+```
+Dog.setLoggingOptions({ skip: ['name'] });
+```
+
+
+---
+
+### Compatible Versions
+Node >= 5
+
+Mongoose >= 4.4
```

Comment:
I'm not sure if we have tested it with the latest versions, would be interesting to run tests for all public mongoose releases starting from 4.4 and see if anything breaks. I think it could break in logger/util.js and [here](https://github.com/teamintrepid/mongoose-event-logger/blob/master/src/plugins/mongoose/logger/plugin.js#L139)

Reply:
You are correct. It breaks for version > 4.6, not due to the line you suspected, but the .save hook. We are unable to monkey patch the save method in this new version due to the way that they have re-implemented the pre and post save hooks. The remove, findOneAndUpdate methods are fine though.

I am looking into it, but it is taking time to figure out. Might need a different solution

Reply:
https://github.com/Automattic/mongoose/issues/3538

See this issue, the creator recommends to not monkey patch save.
I think we should just replace the .save method without another custom one,
for example:
`myModel.saveAndLog();`
Thoughts?